### ISSUE TITLE
Gdal2tiles small non adjacent square issue tiling output with large image

### ISSUE THREAD

**Issue by jkeruzec** [OSGeo/gdal#1980]
I have gdal with gdal2tile, gdal V2.4.1 installed on Centos 7.

I used to tile images of 5MB with resolution like ~ 5000x~5000 in pixels.
Everything was OK

Now I'm starting to tile images with resolution : ~ 10000x~10000 in pixels
Image is still 5MB
But tiling is not working anymore.

I have this kind of output : 
![image](https://user-images.githubusercontent.com/3499632/68134655-bd374e00-ff22-11e9-8956-ba4b9f7f13b5.png)

I already tweaked the GDAL_ALLOW_LARGE_LIBJPEG_MEM_ALLOC to allow bigger image in size.

Can you help ?

Thanks.

**Comment by rouault** [comment]
Please provide an input file and the exact gdal2tiles command line you use

**Comment by jkeruzec** [comment]
Command line : 
`/usr/lib64/python2.7/site-packages/GDAL-2.4.1-py2.7-linux-x86_64.egg/EGG-INFO/scripts/gdal2tiles.py --s_srs EPSG:3857 -q --processes=8 -p raster -z "0-5" -w none %s %s`

Image in attachement
![téléchargement](https://user-images.githubusercontent.com/3499632/68187565-498b5480-ffa7-11e9-92ef-06c10312f793.jpg)

I also did this in python script to handle jpg file by default (found on internet).

```
self.tiledriver = 'JPEG'
--
self.tileext = 'jpg'
```

```
if tile_driver == 'JPEG' and tilebands == 4:
--
tilebands = 3
```
But still, it's working for smaller resolution but not for bigger one's... :/

**Comment by jratike80** [comment]
I don't know why, but it seems to work right if I run your command without `-z "0-5"`.
Please try.

**Comment by jkeruzec** [comment]
Yes, in my particular case, I've added `-z "0-5"` because I had problem with black level zoom ... So I fix something but create another problem :/

**Comment by jratike80** [comment]
I would call it "find", not create.

**Comment by dtwd668** [comment]
Have you solved the problem? I can't deal with  big images too

**Comment by rouault** [comment]
closing as superseded by gdal raster tile
